google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -66,8 +69,45 @@ public FileType getFileType() {
   @NotNull
   @Override
   public ElementPattern<FileContent> createSuitableFilePattern() {
-    return StandardPatterns.or(
-        FileContentPattern.fileContent().withName("app.yaml"),
-        FileContentPattern.fileContent().withName("app.yml"));
+    return new AppEngineFlexPattern().withAppEngineFlexYamlContent();
+  }
+
+  /**
+   * IntelliJ API pattern class that checks for App Engine Flex project file presence and checks it
+   * has required configuration lines to avoid spurious detection.
+   */
+  private static final class AppEngineFlexPattern
+      extends ObjectPattern<FileContent, AppEngineFlexPattern> {
+
+    private AppEngineFlexPattern() {
+      super(FileContent.class);
+    }
+
+    private AppEngineFlexPattern withAppEngineFlexYamlContent() {
+      return with(
+          new PatternCondition<FileContent>("with-appengine-java-flexible") {
+            // app engine flex file names.
```

Comment:
looks good.

nit: remove this comment? its pretty much repeating the variable name

Reply:
agree